src/lib/FieldTypeProcessorRegistry.php

```diff
@@ -32,10 +32,18 @@ public function __construct(array $processors = [])
      * Registers $processor for $fieldTypeIdentifier.
      *
      * @param string $fieldTypeIdentifier
-     * @param \EzSystems\EzPlatformRest\FieldTypeProcessor $processor
+     * @param \EzSystems\EzPlatformRest\FieldTypeProcessorInterface $processor
      */
-    public function registerProcessor($fieldTypeIdentifier, FieldTypeProcessor $processor)
+    public function registerProcessor($fieldTypeIdentifier, /*FieldTypeProcessorInterface*/ $processor)
     {
+        // The FieldTypeProcessorInterface was a late addition to the FieldTypeProcessor class.
+        // The type validation is done this way to insure BC, ie. allowing subclasses of this class
+        // to keep working even if they have the same method overridden with the original signature
+        if (! $processor instanceof FieldTypeProcessorInterface) {
+            throw new \TypeError('Argument 2 passed to ' . get_class($this) . '::registerProcessor() must be an' .
+                ' instance of EzSystems\EzPlatformRest\FieldTypeProcessorInterface, instance of ' . get_class($processor) .
+                ' given');
+        }
```

Comment:
This does not make any sense BC-wise. Even though you defined
```php
public function registerProcessor($fieldTypeIdentifier, /*FieldTypeProcessorInterface*/ $processor)
```
(which is ugly anti-pattern spreading through Symfony), it does not give any value since you're throwing exception here.
Moreover implementation of Field Type Processor is documented as [through extending FieldTypeProcessor base class](https://doc.ibexa.co/en/latest/api/field_type_type_and_value/#extensibility-points), so 
```php
public function registerProcessor($fieldTypeIdentifier, FieldTypeProcessorInterface $processor)
```
is relatively BC-safe and all this spaghetti tricks are not required (though take into the account previous, naming-related, comment).